File: problems/Product of Array Except Self/productExceptSelf.cs
public class Solution {
    public int[] ProductExceptSelf(int[] nums) {
        var n = nums.Length;
        var suffixProd = 1;
        var prefixProds = new int[n];

        prefixProds[0] = 1;

        for (var i = 1; n > i; ++i) {
            prefixProds[i] = prefixProds[i - 1] * nums[i - 1];
        }

        for (var i = n - 1; 0 <= i; --i) {
            prefixProds[i] *= suffixProd;
            suffixProd *= nums[i];
        }

        return prefixProds;
    }
}

File: problems/Product of Array Except Self/productExceptSelf.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int n = nums.length;
        int suffixProd = 1;
        int[] prefixProds = new int[n];

        prefixProds[0] = 1;

        for (int idx = 1; n > idx; ++idx) {
            prefixProds[idx] = prefixProds[idx - 1] * nums[idx - 1];
        }

        for (int idx = n - 1; 0 <= idx; --idx) {
            prefixProds[idx] *= suffixProd;
            suffixProd *= nums[idx];
        }

        return prefixProds;
    }
}
